compiler/kernel/op.go

```diff
@@ -308,6 +310,11 @@ func splitAssignments(assignments []expr.Assignment) (field.List, []expr.Evaluat
 }
 
 func (b *Builder) compileSequential(seq *dag.Sequential, parents []zbuf.Puller) ([]zbuf.Puller, error) {
+	for _, fn := range seq.Funcs {
+		if err := b.compileFunc(fn); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I think we should consider compiling all of the functions together, as in b.compileFuncs(seq.Funcs). That would let us add all of the names to b.funcs before compiling any of the bodies. And that would in turn allow both declaring a caller before a callee and mutual recursion (i.e., a calls b and b calls a).

Reply:
Good idea.